### DISCUSSION TITLE
How to get the context / payload of the existing request during an exception?

### DISCUSSION THREAD

**Discussion by moshevayner** [fastapi/fastapi#6360] [Questions] [answered]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from typing import Generator
import bugsnag
from sqlalchemy.orm import Session


def get_db() -> Generator:
    db = SessionLocal()
    try:
        yield db
    except Exception as e:
        db.rollback()
        bugsnag.notify(e)
        raise
    finally:
        db.close()
```

### Description

I am using [bugsnag's](https://docs.bugsnag.com/platforms/python/asgi/customizing-error-reports/#adding-callbacks) SDK to grab any handled and unhandled exception that occur in my FastAPI api.
I've noticed that when an exception occurs, I don't have the request payload for relevant requests (PUT, POST, etc.).
So I'm trying to figure out how during an exception I can grab the current `Request` object and parse it to extract the desired data.
Something similar to the logic of `get_json` in Flask, so that when I use the `bugsnag.notify()` method, I can also include the incoming request payload. 
You can see in the example code how I use my `get_db` dependency to `yield` a database session and then during every exception that occurs in the code, I rollback my transaction and then want to notify `bugsnag` of the error. 
I'd appreciate any help with this.
Thanks!

### Operating System

macOS

### FastAPI Version

fastapi[all]==0.75.0

### Python Version

3.10.4

**Comment by tasercake** [comment]
Hey there!

Since you say you're using `get_db` with the dependency injection system, you might be able to do the following:


```python
from typing import Generator
import bugsnag
from sqlalchemy.orm import Session

from fastapi import Request

def get_db(request: Request) -> Generator:
    db = SessionLocal()
    try:
        yield db
    except Exception as e:
        db.rollback()
        bugsnag.notify(e)
        raise
    finally:
        db.close()
```

This way, FastAPI automatically knows to inject the current request object into your `get_db` dependency when it's wrapped in `Depends` in a view function.

Alternatively, could also get just the request body with:

```python
from fastapi import Body

def get_db(body = Body(...)) -> Generator:
	...
```

Check out the docs on [sub-dependencies](https://fastapi.tiangolo.com/tutorial/dependencies/sub-dependencies/#second-dependency-dependable-and-dependant) to learn more about how this works.

**Comment by moshevayner** [comment, marked as answer]
Hey @tasercake 
That worked perfectly! Now I can parse the request body and add it to the notification metadata 🙌🏻 

My code now looks like this:
```python
def get_db(request: Request) -> Generator:
    db = SessionLocal()
    try:
        yield db
    except Exception as e:
        db.rollback()
        if asyncio.run(request.body()):
            req_json = asyncio.run(request.json())
            bugsnag.notify(e, metadata={"Request Body": req_json})
        else:
            bugsnag.notify(e)
        raise
    finally:
        db.close()
```

Thank you so much for that suggestion 🙏🏻
